xmppserver/src/main/webapp/available-plugins.jsp

```diff
@@ -144,7 +129,7 @@
     }
 
     function startDownload(url, version, id) {
-        downloader.installPlugin(url, version, id, downloadComplete);
+        downloader.installPlugin('${admin:escapeHTMLTags(webManager.user.username).replaceAll("'", "&quot;")}', url, version, id, downloadComplete);
```

Comment:
This is rather defensive of the username.
I can imagine an LDAP username being pretty complex.
What else?

Also, shouldn't this be sanitised at the point of use, rather than at the callsite, in case its called elsewhere in the future? (Not a strong opinion - just an idea)

Reply:
I'm escaping this here because the username is going to be a _hard-coded_ value in the generated Javascript code. In that sense, I think it _is_ the point of use (the username isn't passed as a referenced value, it's quite literally in the Javascript code. I'm not actually sure if we allow a `'` character in the username, so escaping that (to not mess up the Javascript call may be overly defensive - but I can live with that.